include/aws/s3/private/s3_request_messages.h

```diff
@@ -56,7 +57,8 @@ struct aws_http_message *aws_s3_complete_multipart_message_new(
 
 struct aws_http_message *aws_s3_message_util_copy_http_message(
     struct aws_allocator *allocator,
-    struct aws_http_message *message);
+    struct aws_http_message *message,
+    bool sse);
```

Comment:
Missed this before!  Not important right now, but maybe add a TODO to break out the sse boolean into a flag.  It'll make it more clear when you're using the function (although I see that you annotated it everywhere, which is awesome), and it'll let us add other options in the future.